src/Panel/Ui/FilePreview.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+namespace Kirby\Panel\Ui;
+
+use Kirby\Cms\App;
+use Kirby\Cms\File;
+use Kirby\Panel\Ui\FilePreviews\FileDefaultPreview;
+use Kirby\Toolkit\I18n;
+
+/**
+ * Defines a component that implements a file preview
+ *
+ * @package   Kirby Panel
+ * @author    Nico Hoffmann <nico@getkirby.com>
+ * @link      https://getkirby.com
+ * @copyright Bastian Allgeier
+ * @license   https://getkirby.com/license
+ * @since     5.0.0
+ * @internal
+ */
+abstract class FilePreview extends Component
+{
+	public string $component = 'k-file-default-preview';
+
+	public function __construct(
+		public File $file
+	) {
+	}
+
+	/**
+	 * Returns true if this class should
+	 * handle the preview of this file
+	 */
+	abstract public static function accepts(File $file): bool;
+
+	/**
+	 * Returns detail information about the file
+	 */
+	public function details(): array
+	{
+		return [
+			[
+				'title' => I18n::translate('template'),
+				'text'  => $this->file->template() ?? '—'
+			],
+			[
+				'title' => I18n::translate('mime'),
+				'text'  => $this->file->mime()
+			],
+			[
+				'title' => I18n::translate('url'),
+				'text'  => $this->file->id(),
+				'link'  => $this->file->previewUrl()
+			],
+			[
+				'title' => I18n::translate('size'),
+				'text'  => $this->file->niceSize()
+			],
+		];
+	}
```

Comment:
`Comment`: I think these details are useful for any file preview made, so I put them in the abstract class.